Simplified/NYPLKeychainManager.swift

```diff
@@ -172,38 +174,52 @@ import Foundation
       ]
       let status = SecItemAdd(addQuery as CFDictionary, nil)
       if status != errSecSuccess && status != errSecDuplicateItem {
-        // This is unexpected
-        var errMsg = ""
-        if #available(iOS 11.3, *) {
-          errMsg = (SecCopyErrorMessageString(status, nil) as String?) ?? ""
-        }
-        if errMsg.isEmpty {
-          switch status {
-          case errSecUnimplemented:
-            errMsg = "errSecUnimplemented"
-          case errSecDiskFull:
-            errMsg = "errSecDiskFull"
-          case errSecIO:
-            errMsg = "errSecIO"
-          case errSecOpWr:
-            errMsg = "errSecOpWr"
-          case errSecParam:
-            errMsg = "errSecParam"
-          case errSecWrPerm:
-            errMsg = "errSecWrPerm"
-          case errSecAllocate:
-            errMsg = "errSecAllocate"
-          case errSecUserCanceled:
-            errMsg = "errSecUserCanceled"
-          case errSecBadReq:
-            errMsg = "errSecBadReq"
-          default:
-            errMsg = "Unknown OSStatus: \(status)"
-          }
-        }
-        
-        Log.error(#file, "FeedbookKeyManagement Error: \(errMsg)")
+        logKeychainError(forVendor: vendor.rawValue, status: status, message: "FeedbookKeyManagement Error:")
+      }
+    }
+  }
+    
+  private class func manageFeedbookDrmPrivateKey() {
+    // Request DRM certificates for all vendors
+    for vendor in AudioBookVendors.allCases {
+      vendor.updateDrmCertificate()
+    }
+  }
+    
+  class func logKeychainError(forVendor vendor:String, status: OSStatus, message: String) {
+    // This is unexpected
+    var errMsg = ""
+    if #available(iOS 11.3, *) {
+      errMsg = (SecCopyErrorMessageString(status, nil) as String?) ?? ""
+    }
+    if errMsg.isEmpty {
+      switch status {
+      case errSecUnimplemented:
+        errMsg = "errSecUnimplemented"
+      case errSecDiskFull:
+        errMsg = "errSecDiskFull"
+      case errSecIO:
+        errMsg = "errSecIO"
+      case errSecOpWr:
+        errMsg = "errSecOpWr"
+      case errSecParam:
+        errMsg = "errSecParam"
+      case errSecWrPerm:
+        errMsg = "errSecWrPerm"
+      case errSecAllocate:
+        errMsg = "errSecAllocate"
+      case errSecUserCanceled:
+        errMsg = "errSecUserCanceled"
+      case errSecBadReq:
+        errMsg = "errSecBadReq"
+      default:
+        errMsg = "Unknown OSStatus: \(status)"
       }
     }
+    
+    NYPLErrorLogger.logError(withCode: .keychainItemAddFail,
+                             summary: "Keychan \(errMsg) error for vendor \(vendor)",
```

Comment:
can you remove the `errMsg` from the summary?  The reason is that the summary is used as the title in Crashlytics, so we don't want to have a huge proliferation of error types there. 

It's a difficult compromise, but after browsing those reports a lot I am trying to be more precise in how we log events, they easily become hard to browse.

Reply:
sure, done